nuxeo-services/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/web/common/requestcontroller/filter/NuxeoCorsCsrfFilter.java

```diff
@@ -95,13 +105,57 @@
 
     public static final String ALLOW_NULL_ORIGIN_DEFAULT = "false";
 
+    /** @since 10.3 */
+    public static final String CSRF_TOKEN_NS_PROP = "nuxeo.csrf.token";
+
+    /**
+     * Allows to enforce the use of a CSRF token. Under the {@value #CSRF_TOKEN_NS_PROP} namespace.
+     *
+     * @since 10.3
+     */
+    public static final String CSRF_TOKEN_ENABLED_SUBPROP = "enabled";
+
+    public static final String CSRF_TOKEN_ENABLED_DEFAULT = "false";
+
+    /**
+     * Endpoints for which no CSRF token check is done. Under the {@value #CSRF_TOKEN_NS_PROP} namespace.
+     *
+     * @since 10.3
+     */
+    public static final String CSRF_TOKEN_SKIP_SUBPROP = "skip";
+
+    /** Session attribute in which token is stored. */
+    public static final String CSRF_TOKEN_ATTRIBUTE = "NuxeoCSRFToken";
+
+    /** Request header to request a token or fetch one. */
+    public static final String CSRF_TOKEN_HEADER = "CSRF-Token";
+
+    /** Pseudo-value requesting a new token. */
+    public static final String CSRF_TOKEN_FETCH = "fetch";
+
+    public static final String CSRF_TOKEN_PARAM = "csrf-token";
+
     protected boolean allowNullOrigin;
 
+    protected boolean csrfTokenEnabled;
+
+    protected List<String> csrfTokenSkipPaths;
+
     @Override
     public void init(FilterConfig filterConfig) {
         ConfigurationService configurationService = Framework.getService(ConfigurationService.class);
         allowNullOrigin = Boolean.parseBoolean(
                 configurationService.getProperty(ALLOW_NULL_ORIGIN_PROP, ALLOW_NULL_ORIGIN_DEFAULT));
+        Map<String, Serializable> csrfTokenConfig = configurationService.getProperties(CSRF_TOKEN_NS_PROP);
+        csrfTokenEnabled = Boolean.parseBoolean(StringUtils.defaultString(
+                (String) csrfTokenConfig.get(CSRF_TOKEN_ENABLED_SUBPROP), CSRF_TOKEN_ENABLED_DEFAULT));
+        csrfTokenSkipPaths = new ArrayList<>();
+        Serializable skipPaths = csrfTokenConfig.get(CSRF_TOKEN_SKIP_SUBPROP);
+        if (skipPaths instanceof String[]) {
```

Comment:
It looks like you are turning an array into a List.  Did you consider `Arrays.asList()` ?